### PR TITLE
Add importmap support in generator

### PR DESCRIPTION
**Pull request by megatux**
Rails 7 and up use _importmaps_. This PR adds support for it.

### CODE REVIEW

File: lib/generators/htmx/install_generator.rb [hunk 1]
```diff
@@ -38,7 +63,7 @@ def setup_sprockets
       end
 
       def setup_webpacker
-        `yarn add htmx.org`
+        run 'yarn add htmx.org'
```
Comment (megatux) [inline lib/generators/htmx/install_generator.rb:66]: the #run helper is from Thor gem (thor-1.2.2/lib/thor/actions.rb
). Has some nice stuff.

File: lib/generators/htmx/install_generator.rb [hunk 2]
```diff
@@ -23,10 +28,30 @@ def webpacker?
...
+      def importmap?
+        !!defined?(Importmap)
+      end
+
       def manifest(javascript_dir)
         Pathname.new(destination_root).join(javascript_dir, 'application.js')
       end
 
+      def setup_importmap
+        run 'bin/importmap pin htmx.org'
```
[the remark this answers is not in the document]
Comment (JulianPasquale) [reply lib/generators/htmx/install_generator.rb:44]: I think this is fine

File: spec/generators/install_generator_spec.rb [hunk 3]
```diff
@@ -42,10 +48,11 @@
   context 'with Webpacker' do
     before do
       stub_const('Webpacker', Module.new)
+      hide_const('Sprockets')
+      hide_const('Importmap')
 
-      Htmx::Generators::InstallGenerator
-        .any_instance
-        .stub(:webpack_source_path)
+      allow_any_instance_of(Htmx::Generators::InstallGenerator)
```
Comment (megatux) [inline spec/generators/install_generator_spec.rb:54]: fix deprecation warning on old RSpec syntax

File: spec/generators/install_generator_spec.rb [hunk 4]
```diff
@@ -42,10 +48,12 @@
   context 'with Webpacker' do
     before do
       stub_const('Webpacker', Module.new)
+      hide_const('Sprockets')
+      hide_const('Importmap')
 
-      Htmx::Generators::InstallGenerator
-        .any_instance
-        .stub(:webpack_source_path)
+      expect_any_instance_of(Htmx::Generators::InstallGenerator)
```
Comment (megatux) [inline spec/generators/install_generator_spec.rb:54]: fix deprecation warning on old Rspec syntax

File: lib/generators/htmx/install_generator.rb [hunk 5]
```diff
@@ -5,15 +5,20 @@ module Generators
     class InstallGenerator < ::Rails::Generators::Base
       WEBPACKER_SETUP = "require('htmx.org')\n"
       SPROCKETS_SETUP = "//= require htmx\n"
+      IMPORTMAP_SETUP = "import \"htmx.org\"\n"
 
       desc 'Prep application.js to include HTMX installation for Webpacker or Sprockets'
```
[the remark this answers is not in the document]
Comment (megatux) [reply lib/generators/htmx/install_generator.rb]: done

**Review by JulianPasquale — APPROVED** [review]

**Pull request merged**